Unstructured-IO/unstructured-js-client
# chore/set pdf page splitting to true by default

Mirror of https://github.com/Unstructured-IO/unstructured-python-client/pull/118
* Set the split_pdf_page default to true and run `make client-generate` locally.
* Update the readme, add another reference back to our docs, bring back some autogenerated sections like in the python repo
* Change some warning logs to info. The user should not be warned about default behavior for non pdf files

# Testing
Use the client locally and verify that split mode is the default, and that the dev experience is good

* Create a new test dir and run `npm init -y; npm install typescript tsx`
* Check out this branch and install from your test dir: `npm i file:~/repos/unstructured-js-client`
* Run this sample script. Try some different files in and verify that the logging and results look acceptable.

`npx tsx unstructured.ts`


```
import { UnstructuredClient } from "unstructured-client";
import { PartitionResponse } from "unstructured-client/sdk/models/operations";
import { Strategy } from "unstructured-client/sdk/models/shared";
import * as fs from "fs";

const key = "free-api-key";

const client = new UnstructuredClient({
    security: {
        apiKeyAuth: key,
    },
});

const filename = "fake-html.html";
const data = fs.readFileSync(filename);

client.general.partition({
    partitionParameters: {
        files: {
            content: data,
            fileName: filename,
        },
        strategy: Strategy.Auto,
    }
}).then((res: PartitionResponse) => {
    if (res.statusCode == 200) {
        console.log(res.elements);
    }
}).catch((e) => {
    if (e.statusCode) {
      console.log(e.statusCode);
      console.log(e.body);
    } else {
      console.log(e);
    }
});
```
